Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py

```diff
@@ -0,0 +1,2769 @@
+        additional_headers = {'Range': range_} if not domain_id and range_ else None
+        return self.http_request(
+            method='GET',
+            url_suffix=f'/config/domain_management/domains{id_suffix}',
+            additional_headers=additional_headers,
+            params=params
+        )
+
+    def indicators_upload(self, ref_name: str, indicators: Any, fields: Optional[str] = None):
+        headers = {
+            'Content-Type': 'application/json'
+        }
+        if fields:
+            headers['fields'] = fields
+        return self.http_request(
+            method='POST',
+            url_suffix=f'/reference_data/sets/bulk_load/{ref_name}',
+            json_data=indicators,
+            additional_headers=headers
+        )
+
+    def geolocations_for_ip(self, filter_: Optional[str] = None, fields: Optional[str] = None):
+        return self.http_request(
+            method='GET',
+            url_suffix='/services/geolocations',
+            params=assign_params(filter=filter_, fields=fields)
+        )
+
+    def log_sources_list(self, qrd_encryption_algorithm: str, qrd_encryption_password: str,
+                         range_: str, filter_: Optional[str] = None, fields: Optional[str] = None):
+        return self.http_request(
+            method='GET',
+            url_suffix='/config/event_sources/log_source_management/log_sources',
+            params=assign_params(filter=filter_, fields=fields),
+            additional_headers={
+                'x-qrd-encryption-algorithm': qrd_encryption_algorithm,
+                'x-qrd-encryption-password': qrd_encryption_password,
+                'Range': range_
+            }
+        )
+
+    def custom_properties(self, range_: Optional[str] = None, filter_: Optional[str] = None,
+                          fields: Optional[str] = None):
+        return self.http_request(
+            method='GET',
+            url_suffix='/config/event_sources/custom_properties/regex_properties',
+            params=assign_params(filter=filter_, fields=fields),
+            additional_headers={'Range': range_} if range_ else None
+        )
+
+    def offense_types(self, filter_: Optional[str] = None, fields: Optional[str] = None):
+        return self.http_request(
+            method='GET',
+            url_suffix='/siem/offense_types',
+            params=assign_params(filter=filter_, fields=fields)
+        )
+
+    def get_addresses(self, address_suffix: str, filter_: Optional[str] = None, fields: Optional[str] = None):
+        return self.http_request(
+            method='GET',
+            url_suffix=f'/siem/{address_suffix}',
+            params=assign_params(filter=filter_, fields=fields)
+        )
+
+    def test_connection(self):
+        """
+        Test connection with databases (should always be up)
+        """
+        self.http_request(method='GET', url_suffix='/ariel/databases')
```

Comment:
@DeanArbel 
wouldn't offense 0-0 return an empty list?
if so, maybe reference-set 0-0 or something like that might not fail and can be used instead?

Reply:
It's ok that it'll return an empty list since you are just testing the connection and authorization - i.e. to check that the user configured the instance correctly and has sufficient authorization.